Example/Tests/DdSdkTests.swift

```diff
@@ -114,35 +114,156 @@ internal class DdSdkTests: XCTestCase {
+            env: "env",
+            applicationId: "app-id",
+            nativeCrashReportEnabled: true,
+            sampleRate: 75.0,
+            site: "US5",
+            trackingConsent: "pending",
+            additionalConfig: nil
+        )
+
+        let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
+
+        XCTAssertEqual(ddConfig.datadogEndpoint, .us5)
+    }
+
+    func testBuildConfigurationUS1FEDEndpoint() {
+        let configuration = DdSdkConfiguration(
+            clientToken: "client-token",
+            env: "env",
+            applicationId: "app-id",
+            nativeCrashReportEnabled: true,
+            sampleRate: 75.0,
+            site: "US1_FED",
+            trackingConsent: "pending",
+            additionalConfig: nil
+        )
+
+        let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
+
+        XCTAssertEqual(ddConfig.datadogEndpoint, .us1_fed)
     }
 
     func testBuildConfigurationGOVEndpoint() {
-        let configuration = DdSdkConfiguration(clientToken: "client-token", env: "env", applicationId: "app-id", nativeCrashReportEnabled: true, sampleRate: 75.0, site: "GOV", trackingConsent: "pending", additionalConfig: nil)
+        let configuration = DdSdkConfiguration(
+            clientToken: "client-token",
+            env: "env",
+            applicationId: "app-id",
+            nativeCrashReportEnabled: true,
+            sampleRate: 75.0,
+            site: "GOV",
+            trackingConsent: "pending",
+            additionalConfig: nil
+        )
+
+        let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
+
+        XCTAssertEqual(ddConfig.datadogEndpoint, .us1_fed)
+    }
+
+    func testBuildConfigurationEUEndpoint() {
+        let configuration = DdSdkConfiguration(
+            clientToken: "client-token",
+            env: "env",
+            applicationId: "app-id",
+            nativeCrashReportEnabled: true,
+            sampleRate: 75.0,
+            site: "EU",
+            trackingConsent: "pending",
+            additionalConfig: nil
+        )
+
+        let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
+
+        XCTAssertEqual(ddConfig.datadogEndpoint, .eu1)
+    }
+
+    func testBuildConfigurationEU1Endpoint() {
+        let configuration = DdSdkConfiguration(
+            clientToken: "client-token",
+            env: "env",
+            applicationId: "app-id",
+            nativeCrashReportEnabled: true,
+            sampleRate: 75.0,
+            site: "EU1",
+            trackingConsent: "pending",
+            additionalConfig: nil
+        )
 
         let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
 
-        XCTAssertEqual(ddConfig.datadogEndpoint, Datadog.Configuration.DatadogEndpoint.gov)
+        XCTAssertEqual(ddConfig.datadogEndpoint, .eu1)
     }
```

Comment:
nit: it's quite repetitive already, so to make it easier for next contributions we could introduce partial mock:
```swift
func testBuildConfigurationEU1Endpoint() {
   let configuration: DdSdkConfiguration = .mockWith(site: "EU1")
   let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
   XCTAssertEqual(ddConfig.datadogEndpoint, .eu1)
}
```